webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/campaigns/CampaignOrchestrationLogic.scala

```diff
@@ -265,18 +272,26 @@ class CampaignOrchestrationLogic(effects: CampaignOrchestrationEffects) {
         }
 
       case PostHooksType =>
+        val state = event.state.asInstanceOf[PostHooks]
         for {
           res <- effects.runPostHooks(campaign, event)
         } yield {
           // hooks are special: we don't call per-campaign type handler and we need to first save before change state/update
-          val nextState =
-            res.results.collectFirst { case r if isError(r) => Failure("post-hooks were in error", r.stderr) }.getOrElse(Finished)
-          EventOrchestration.SaveThenUpdateAndQueue(event.copy(state = PostHooks(res)), nextState)
+          val nextState = {
+            val optError = res.results.collectFirst { case r if isError(r) => Failure("post-hooks were in error", r.stderr) }
+            (optError, state.nextState) match {
+              case (None, FailureType)     => Failure("pre-hooks were in error and post-hooks completed successfully", "")
+              case (None, s)               => getDefault(s)
+              case (Some(f1), FailureType) => Failure("pre-hooks and post-hooks were in error. Last error:", f1.cause)
+              case (Some(f1), _)           => f1
```

Comment:
this logic is just here to put something relevant in the failure message. We don't keep the whole error, so we don't know what hook errored at that point. 
In the futur, we will just have the whole history in the campaign event UI and the user will be able to see what broke.